aesara/tensor/math_opt.py

```diff
@@ -1794,8 +1794,9 @@ def local_mul_zero(fgraph, node):
                 value = get_scalar_constant_value(i)
             except NotScalarConstantError:
                 continue
-            # print 'MUL by value', value, node.inputs
-            if value == 0:
+            # Print 'MUL by value', value, node.inputs.
+            # We use value.any() to avoid ambiguous array with more than one element.
+            if value.any() == 0:
```

Comment:
I think this condition is supposed to actually evaluate to `False` if `value` is an array, because we don't want to "optimize" `a * [1, 0, 2]` to `0`.

Reply:
Thanks for the clarification!

I changed it to the following

From 

```python
            # Print 'MUL by value', value, node.inputs.
            # We use value.any() to avoid ambiguous array with more than one element.
            if value.any() == 0:
```

To 

```python
            # Validate code to optimize operation to result in array.
            # E.g. We do not want't to "optimize" `a * [1, 0, 2]`` to 0.
            if value.any() is False:
```
______

I get this unrelated error, unsure if it's my Aesara version

```
FAILED tests/tensor/test_math_opt.py::TestLocalUselessElemwiseComparison::test_local_useless_elemwise_comparison - Exception: ('The foll...
```

Reply:
I'm very sure that we have to keep this exactly as it was before:
```
# print 'MUL by value', value, node.inputs
if value == 0:
```